### PR TITLE
Load ".enc" files encrypted with Sekrets

### PR DESCRIPTION
**Pull request by spaghetticode**
Dotenv is great, I've been using it for long in all my projects. But one feature I miss is the possibility to version-control `.env` files: as you may know saving configuration secrets/passwords inside git repos is a big no-no... as long as they're clear. We avoid to commit even `.env.development` files for security reasons. 

Saving encrypted configuration would vastly help our distributed teams, we would just need to pass along the decryption key instead of a whole config file which, more often than not, is not complete and up to current development status, because each developer ends up having/receiving/spreading a different version of the file. That's messy.

So I started playing with both `dotenv` and the `sekrets` gem to make them work together. I can now load encrypted dotenv files also with ".enc" extension, which can now be safely commited into the repository. 

Of course the code is quite rough, I am not even sure that dotenv is the right place to add this enhancement (maybe a new `dotenv-sekrets` gem?) so I am submitting the PR just to start a conversation about the encryption topic.

### CODE REVIEW

**Comment by bkeepers** [comment]
Hey @spaghetticode, thanks for the pull request!

> …I am not even sure that dotenv is the right place to add this enhancement (maybe a new `dotenv-sekrets` gem?) so I am submitting the PR just to start a conversation about the encryption topic.

This is a cool feature, but I think publishing it as a separate gem is the way to go, at least initially. You could re-use parts of `dotenv`, and I would even be up for making some changes to make it easier for you to extend it.

Sound ok?

File: lib/dotenv/environment.rb [hunk 1]
```diff
@@ -14,7 +14,15 @@ def load
     end
 
     def read
-      File.open(@filename, "rb:bom|utf-8", &:read)
+      if defined? Sekrets
+        if @filename =~ /\.enc$/
+          Sekrets.read(@filename) || raise(Errno::ENOENT)
+        else
+          File.open(@filename, "rb:bom|utf-8", &:read)
```
Comment (siegy22) [inline lib/dotenv/environment.rb]: you can use `File.read` so you don't have to pass `&:read`.

**Comment by spaghetticode** [comment]
@bkeepers Of course it sounds ok. I think that once you provide some hooks to the existing code then my job (or anyone else who plans to extend dotenv functionalities) would be more straightforward, I will be waiting for updates, thank you!

**Comment by spaghetticode** [comment]
Meanwhile I've released an external gem for this feature: https://github.com/mikamai/dotenv_sekrets.

I'll be waiting for any update from you, meanwhile I think it's ok to close this PR without merging the code, please just let me know if it's something I should do myself.

**Comment by bkeepers** [comment]
Hey @spaghetticode, that's awesome!

I'm not actively working on dotenv at the moment, but would be happy to review pull requests that make the desired changes.

**Pull request closed without merging**